CMakeLists.txt

```diff
@@ -37,8 +37,21 @@ endif()
 # Boost package
 ###############################################################################
 
+set(Boost_COMPONENTS coroutine context system thread)
+
+# If on Windows, rather than be annoyed by constant failure to find DLL errors
+# during test debugging simply force all static linking of Boost libraries
+if(WIN32 AND NOT MINGW)
+  set(Boost_USE_STATIC_LIBS ON)
+else()
+  set(Boost_COMPONENTS ${Boost_COMPONENTS} unit_test_framework)
```

Comment:
Aha, so this change could be read together with the `test/CMakeLists.txt` file. What I do not understand is that here you add the `unit_test_framework` component to all but Visual Studio. Why does, say, Linux need to find the component, but Visual Studio does not?

Looking over the checks again, I would prefer if we define a CMake variable or option to specify to use of header-only Boost.Test, and handle the MinGW workaround here. Then later we can use this variable to determine whether or not to use the `unit_test_framework` component and the `BOOST_TEST_DYN_LINK` macro.